Fix build warnings

I'm getting a fair amount of warnings when building the library. They should all be simple to fix. @olikraus  would you be willing to merge a PR if I work on it?

```
  CC	../ucglib/csrc/ucg_rotate.c
../ucglib/csrc/ucg_rotate.c:68:11: warning: no previous prototype for 'ucg_dev_rotate90' [-Wmissing-prototypes]
 ucg_int_t ucg_dev_rotate90(ucg_t *ucg, ucg_int_t msg, void *data)
           ^~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_rotate.c:149:11: warning: no previous prototype for 'ucg_dev_rotate180' [-Wmissing-prototypes]
 ucg_int_t ucg_dev_rotate180(ucg_t *ucg, ucg_int_t msg, void *data)
           ^~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_rotate.c:208:11: warning: no previous prototype for 'ucg_dev_rotate270' [-Wmissing-prototypes]
 ucg_int_t ucg_dev_rotate270(ucg_t *ucg, ucg_int_t msg, void *data)
           ^~~~~~~~~~~~~~~~~
  CC	../ucglib/csrc/ucg_com_msg_api.c
../ucglib/csrc/ucg_com_msg_api.c:38:9: warning: no previous prototype for 'ucg_com_template_cb' [-Wmissing-prototypes]
 int16_t ucg_com_template_cb(ucg_t *ucg, int16_t msg, uint32_t arg, uint8_t *data)
         ^~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_com_msg_api.c: In function 'ucg_com_template_cb':
../ucglib/csrc/ucg_com_msg_api.c:38:36: warning: unused parameter 'ucg' [-Wunused-parameter]
 int16_t ucg_com_template_cb(ucg_t *ucg, int16_t msg, uint32_t arg, uint8_t *data)
                                    ^~~
../ucglib/csrc/ucg_com_msg_api.c:38:63: warning: unused parameter 'arg' [-Wunused-parameter]
 int16_t ucg_com_template_cb(ucg_t *ucg, int16_t msg, uint32_t arg, uint8_t *data)
                                                               ^~~
../ucglib/csrc/ucg_com_msg_api.c:38:77: warning: unused parameter 'data' [-Wunused-parameter]
 int16_t ucg_com_template_cb(ucg_t *ucg, int16_t msg, uint32_t arg, uint8_t *data)
                                                                             ^~~~
../ucglib/csrc/ucg_com_msg_api.c: At top level:
../ucglib/csrc/ucg_com_msg_api.c:182:6: warning: no previous prototype for 'ucg_com_SendStringP' [-Wmissing-prototypes]
 void ucg_com_SendStringP(ucg_t *ucg, uint16_t cnt, const ucg_pgm_uint8_t *byte_ptr)
      ^~~~~~~~~~~~~~~~~~~
  CC	../ucglib/csrc/ucg_dev_ic_hx8352c.c
../ucglib/csrc/ucg_dev_ic_hx8352c.c:140:11: warning: no previous prototype for 'ucg_handle_hx8352c_l90fx' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_hx8352c_l90fx(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_hx8352c.c:180:11: warning: no previous prototype for 'ucg_handle_hx8352c_l90tc' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_hx8352c_l90tc(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_hx8352c.c:278:11: warning: no previous prototype for 'ucg_handle_hx8352c_l90se' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_hx8352c_l90se(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_hx8352c.c: In function 'ucg_handle_hx8352c_l90se':
../ucglib/csrc/ucg_dev_ic_hx8352c.c:306:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i, j;
               ^
../ucglib/csrc/ucg_dev_ic_hx8352c.c:280:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_hx8352c.c:306:18: warning: variable 'j' set but not used [-Wunused-but-set-variable]
     ucg_int_t i, j;
                  ^
../ucglib/csrc/ucg_dev_ic_hx8352c.c: In function 'ucg_ext_hx8352c_18':
../ucglib/csrc/ucg_dev_ic_hx8352c.c:404:63: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_hx8352c_18(ucg_t *ucg, ucg_int_t msg, void *data)
                                                               ^~~~
  CC	../ucglib/csrc/ucg_init.c
../ucglib/csrc/ucg_init.c:45:6: warning: no previous prototype for 'ucg_init_struct' [-Wmissing-prototypes]
 void ucg_init_struct(ucg_t *ucg)
      ^~~~~~~~~~~~~~~
  CC	../ucglib/csrc/ucg_dev_ic_pcf8833.c
../ucglib/csrc/ucg_dev_ic_pcf8833.c:149:11: warning: no previous prototype for 'ucg_handle_pcf8833_l90fx' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_pcf8833_l90fx(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_pcf8833.c:192:11: warning: no previous prototype for 'ucg_handle_pcf8833_l90tc' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_pcf8833_l90tc(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_pcf8833.c:277:11: warning: no previous prototype for 'ucg_handle_pcf8833_l90se' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_pcf8833_l90se(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_pcf8833.c: In function 'ucg_handle_pcf8833_l90se':
../ucglib/csrc/ucg_dev_ic_pcf8833.c:294:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i;
               ^
../ucglib/csrc/ucg_dev_ic_pcf8833.c:279:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_pcf8833.c: In function 'ucg_ext_pcf8833_16':
../ucglib/csrc/ucg_dev_ic_pcf8833.c:399:63: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_pcf8833_16(ucg_t *ucg, ucg_int_t msg, void *data)
                                                               ^~~~
  CC	../ucglib/csrc/ucg_dev_tft_240x320_ili9341.c
  CC	../ucglib/csrc/ucg_dev_oled_128x128_ft.c
  CC	../ucglib/csrc/ucg_vector_font_data.c
  CC	../ucglib/csrc/ucg_box.c
  CC	../ucglib/csrc/ucg_clip.c
  CC	../ucglib/csrc/ucg_pixel.c
  CC	../ucglib/csrc/ucg_dev_tft_240x320_ili9325_spi.c
  CC	../ucglib/csrc/ucg_dev_tft_240x320_ssd1289.c
  CC	../ucglib/csrc/ucg_dev_ic_ili9341.c
../ucglib/csrc/ucg_dev_ic_ili9341.c:123:11: warning: no previous prototype for 'ucg_handle_ili9341_l90fx' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ili9341_l90fx(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ili9341.c:167:11: warning: no previous prototype for 'ucg_handle_ili9341_l90tc' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ili9341_l90tc(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ili9341.c:255:11: warning: no previous prototype for 'ucg_handle_ili9341_l90se' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ili9341_l90se(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ili9341.c: In function 'ucg_handle_ili9341_l90se':
../ucglib/csrc/ucg_dev_ic_ili9341.c:272:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i;
               ^
../ucglib/csrc/ucg_dev_ic_ili9341.c:257:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_ili9341.c: In function 'ucg_ext_ili9341_18':
../ucglib/csrc/ucg_dev_ic_ili9341.c:372:63: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_ili9341_18(ucg_t *ucg, ucg_int_t msg, void *data)
                                                               ^~~~
  CC	../ucglib/csrc/ucg_scale.c
../ucglib/csrc/ucg_scale.c:96:11: warning: no previous prototype for 'ucg_dev_scale2x2' [-Wmissing-prototypes]
 ucg_int_t ucg_dev_scale2x2(ucg_t *ucg, ucg_int_t msg, void *data)
           ^~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_scale.c:52:14: warning: 'ucg_scale_2x2_send_next_half_byte' defined but not used [-Wunused-function]
  static void ucg_scale_2x2_send_next_half_byte(ucg_t *ucg, ucg_xy_t *xy, ucg_int_t msg, ucg_int_t len, ucg_int_t dir, uint8_t b)
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CC	../ucglib/csrc/ucg_dev_ic_ssd1351.c
../ucglib/csrc/ucg_dev_ic_ssd1351.c:64:11: warning: no previous prototype for 'ucg_handle_ssd1351_l90fx' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ssd1351_l90fx(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ssd1351.c:123:11: warning: no previous prototype for 'ucg_handle_ssd1351_l90tc' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ssd1351_l90tc(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ssd1351.c:232:11: warning: no previous prototype for 'ucg_handle_ssd1351_l90se' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ssd1351_l90se(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ssd1351.c: In function 'ucg_handle_ssd1351_l90se':
../ucglib/csrc/ucg_dev_ic_ssd1351.c:261:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i, j;
               ^
../ucglib/csrc/ucg_dev_ic_ssd1351.c:234:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_ssd1351.c: In function 'ucg_ext_ssd1351_18':
../ucglib/csrc/ucg_dev_ic_ssd1351.c:362:63: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_ssd1351_18(ucg_t *ucg, ucg_int_t msg, void *data)
                                                               ^~~~
  CC	../ucglib/csrc/ucg_dev_ic_ld50t6160.c
../ucglib/csrc/ucg_dev_ic_ld50t6160.c:226:11: warning: no previous prototype for 'ucg_handle_ld50t6160_l90se' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ld50t6160_l90se(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ld50t6160.c: In function 'ucg_handle_ld50t6160_l90se':
../ucglib/csrc/ucg_dev_ic_ld50t6160.c:243:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i;
               ^
../ucglib/csrc/ucg_dev_ic_ld50t6160.c:228:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_ld50t6160.c: In function 'ucg_ext_ld50t6160_18':
../ucglib/csrc/ucg_dev_ic_ld50t6160.c:348:65: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_ld50t6160_18(ucg_t *ucg, ucg_int_t msg, void *data)
                                                                 ^~~~
  CC	../ucglib/csrc/ucg_dev_tft_132x132_pcf8833.c
  CC	../ucglib/csrc/ucg_dev_ic_ili9325.c
../ucglib/csrc/ucg_dev_ic_ili9325.c:175:11: warning: no previous prototype for 'xxxxxx_ucg_handle_ili9325_l90tc' [-Wmissing-prototypes]
 ucg_int_t xxxxxx_ucg_handle_ili9325_l90tc(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ili9325.c: In function 'ucg_handle_ili9325_l90se':
../ucglib/csrc/ucg_dev_ic_ili9325.c:387:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i;
               ^
../ucglib/csrc/ucg_dev_ic_ili9325.c:373:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_ili9325.c: In function 'ucg_ext_ili9325_18':
../ucglib/csrc/ucg_dev_ic_ili9325.c:474:63: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_ili9325_18(ucg_t *ucg, ucg_int_t msg, void *data)
                                                               ^~~~
At top level:
../ucglib/csrc/ucg_dev_ic_ili9325.c:241:18: warning: 'ucg_handle_ili9325_l90tc' defined but not used [-Wunused-function]
 static ucg_int_t ucg_handle_ili9325_l90tc(ucg_t *ucg)
                  ^~~~~~~~~~~~~~~~~~~~~~~~
  CC	../ucglib/csrc/ucg_dev_ic_ili9325_spi.c
../ucglib/csrc/ucg_dev_ic_ili9325_spi.c: In function 'ucg_handle_ili9325_l90se':
../ucglib/csrc/ucg_dev_ic_ili9325_spi.c:388:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i;
               ^
../ucglib/csrc/ucg_dev_ic_ili9325_spi.c:374:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_ili9325_spi.c: In function 'ucg_ext_ili9325_spi_18':
../ucglib/csrc/ucg_dev_ic_ili9325_spi.c:474:67: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_ili9325_spi_18(ucg_t *ucg, ucg_int_t msg, void *data)
                                                                   ^~~~
At top level:
../ucglib/csrc/ucg_dev_ic_ili9325_spi.c:242:18: warning: 'ucg_handle_ili9325_l90tc' defined but not used [-Wunused-function]
 static ucg_int_t ucg_handle_ili9325_l90tc(ucg_t *ucg)
                  ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ili9325_spi.c:176:18: warning: 'xxxxxx_ucg_handle_ili9325_l90tc' defined but not used [-Wunused-function]
 static ucg_int_t xxxxxx_ucg_handle_ili9325_l90tc(ucg_t *ucg)
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CC	../ucglib/csrc/ucg_bitmap.c
  CC	../ucglib/csrc/ucg_dev_ic_ssd1289.c
  CC	../ucglib/csrc/ucg_polygon.c
../ucglib/csrc/ucg_polygon.c: In function 'pg_AddPolygonXY':
../ucglib/csrc/ucg_polygon.c:277:44: warning: unused parameter 'ucg' [-Wunused-parameter]
 void pg_AddPolygonXY(pg_struct *pg, ucg_t *ucg, int16_t x, int16_t y)
                                            ^~~
  CC	../ucglib/csrc/ucg_line.c
  CC	../ucglib/csrc/ucg_dev_tft_320x480_ili9486.c
  CC	../ucglib/csrc/ucg_dev_ic_st7735.c
../ucglib/csrc/ucg_dev_ic_st7735.c:123:11: warning: no previous prototype for 'ucg_handle_st7735_l90fx' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_st7735_l90fx(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_st7735.c:167:11: warning: no previous prototype for 'ucg_handle_st7735_l90tc' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_st7735_l90tc(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_st7735.c:255:11: warning: no previous prototype for 'ucg_handle_st7735_l90se' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_st7735_l90se(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_st7735.c: In function 'ucg_handle_st7735_l90se':
../ucglib/csrc/ucg_dev_ic_st7735.c:272:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i;
               ^
../ucglib/csrc/ucg_dev_ic_st7735.c:257:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_st7735.c: In function 'ucg_ext_st7735_18':
../ucglib/csrc/ucg_dev_ic_st7735.c:373:62: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_st7735_18(ucg_t *ucg, ucg_int_t msg, void *data)
                                                              ^~~~
  CC	../ucglib/csrc/ucg_dev_msg_api.c
  CC	../ucglib/csrc/ucg_dev_oled_128x128_ilsoft.c
  CC	../ucglib/csrc/ucg_pixel_font_data.c
  CC	../ucglib/csrc/ucg_circle.c
../ucglib/csrc/ucg_circle.c:73:6: warning: no previous prototype for 'ucg_draw_circle' [-Wmissing-prototypes]
 void ucg_draw_circle(ucg_t *ucg, ucg_int_t x0, ucg_int_t y0, ucg_int_t rad, uint8_t option)
      ^~~~~~~~~~~~~~~
../ucglib/csrc/ucg_circle.c:162:6: warning: no previous prototype for 'ucg_draw_disc' [-Wmissing-prototypes]
 void ucg_draw_disc(ucg_t *ucg, ucg_int_t x0, ucg_int_t y0, ucg_int_t rad, uint8_t option)
      ^~~~~~~~~~~~~
  CC	../ucglib/csrc/ucg_dev_tft_128x160_st7735.c
  CC	../ucglib/csrc/ucg_dev_ic_ili9163.c
../ucglib/csrc/ucg_dev_ic_ili9163.c:125:11: warning: no previous prototype for 'ucg_handle_ili9163_l90fx' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ili9163_l90fx(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ili9163.c:176:11: warning: no previous prototype for 'ucg_handle_ili9163_l90tc' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ili9163_l90tc(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ili9163.c:267:11: warning: no previous prototype for 'ucg_handle_ili9163_l90se' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ili9163_l90se(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ili9163.c: In function 'ucg_handle_ili9163_l90se':
../ucglib/csrc/ucg_dev_ic_ili9163.c:284:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i;
               ^
../ucglib/csrc/ucg_dev_ic_ili9163.c:269:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_ili9163.c: In function 'ucg_ext_ili9163_18':
../ucglib/csrc/ucg_dev_ic_ili9163.c:392:63: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_ili9163_18(ucg_t *ucg, ucg_int_t msg, void *data)
                                                               ^~~~
  CC	../ucglib/csrc/ucg_font.c
../ucglib/csrc/ucg_font.c:158:6: warning: no previous prototype for 'ucg_read_font_info' [-Wmissing-prototypes]
 void ucg_read_font_info(ucg_font_info_t *font_info, const ucg_fntpgm_uint8_t *font)
      ^~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:277:8: warning: no previous prototype for 'ucg_font_GetLowerGDescent' [-Wmissing-prototypes]
 int8_t ucg_font_GetLowerGDescent(const void *font)
        ^~~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:356:9: warning: redundant redeclaration of 'ucg_GetFontCapitalAHeight' [-Wredundant-decls]
 uint8_t ucg_GetFontCapitalAHeight(ucg_t *ucg) UCG_NOINLINE;
         ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../ucglib/csrc/ucg_font.c:38:0:
../ucglib/csrc/ucg.h:731:9: note: previous declaration of 'ucg_GetFontCapitalAHeight' was here
 uint8_t ucg_GetFontCapitalAHeight(ucg_t *ucg) UCG_NOINLINE;
         ^~~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:367:9: warning: no previous prototype for 'ucg_font_decode_get_unsigned_bits' [-Wmissing-prototypes]
 uint8_t ucg_font_decode_get_unsigned_bits(ucg_font_decode_t *f, uint8_t cnt)
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:412:8: warning: no previous prototype for 'ucg_font_decode_get_signed_bits' [-Wmissing-prototypes]
 int8_t ucg_font_decode_get_signed_bits(ucg_font_decode_t *f, uint8_t cnt)
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:562:6: warning: no previous prototype for 'ucg_font_decode_len' [-Wmissing-prototypes]
 void ucg_font_decode_len(ucg_t *ucg, uint8_t len, uint8_t is_foreground)
      ^~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:670:8: warning: no previous prototype for 'ucg_font_decode_glyph' [-Wmissing-prototypes]
 int8_t ucg_font_decode_glyph(ucg_t *ucg, const uint8_t *glyph_data)
        ^~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:721:16: warning: no previous prototype for 'ucg_font_get_glyph_data' [-Wmissing-prototypes]
 const uint8_t *ucg_font_get_glyph_data(ucg_t *ucg, uint8_t encoding)
                ^~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:748:11: warning: no previous prototype for 'ucg_font_draw_glyph' [-Wmissing-prototypes]
 ucg_int_t ucg_font_draw_glyph(ucg_t *ucg, ucg_int_t x, ucg_int_t y, uint8_t dir, uint8_t encoding)
           ^~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:864:6: warning: no previous prototype for 'ucg_UpdateRefHeight' [-Wmissing-prototypes]
 void ucg_UpdateRefHeight(ucg_t *ucg)
      ^~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:910:11: warning: no previous prototype for 'ucg_font_calc_vref_font' [-Wmissing-prototypes]
 ucg_int_t ucg_font_calc_vref_font(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c: In function 'ucg_font_calc_vref_font':
../ucglib/csrc/ucg_font.c:910:42: warning: unused parameter 'ucg' [-Wunused-parameter]
 ucg_int_t ucg_font_calc_vref_font(ucg_t *ucg)
                                          ^~~
../ucglib/csrc/ucg_font.c: At top level:
../ucglib/csrc/ucg_font.c:921:11: warning: no previous prototype for 'ucg_font_calc_vref_bottom' [-Wmissing-prototypes]
 ucg_int_t ucg_font_calc_vref_bottom(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:931:11: warning: no previous prototype for 'ucg_font_calc_vref_top' [-Wmissing-prototypes]
 ucg_int_t ucg_font_calc_vref_top(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_font.c:945:11: warning: no previous prototype for 'ucg_font_calc_vref_center' [-Wmissing-prototypes]
 ucg_int_t ucg_font_calc_vref_center(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~~
  CC	../ucglib/csrc/ucg_dev_oled_128x128_univision.c
  CC	../ucglib/csrc/ucg_dev_oled_96x64_univision.c
  CC	../ucglib/csrc/ucg_dev_tft_240x400_hx8352c.c
  CC	../ucglib/csrc/ucg_dev_tft_128x128_ili9163.c
  CC	../ucglib/csrc/ucg_dev_default_cb.c
../ucglib/csrc/ucg_dev_default_cb.c: In function 'ucg_ext_none':
../ucglib/csrc/ucg_dev_default_cb.c:60:31: warning: unused parameter 'ucg' [-Wunused-parameter]
 ucg_int_t ucg_ext_none(ucg_t *ucg, ucg_int_t msg, void *data)
                               ^~~
../ucglib/csrc/ucg_dev_default_cb.c:60:46: warning: unused parameter 'msg' [-Wunused-parameter]
 ucg_int_t ucg_ext_none(ucg_t *ucg, ucg_int_t msg, void *data)
                                              ^~~
../ucglib/csrc/ucg_dev_default_cb.c:60:57: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_none(ucg_t *ucg, ucg_int_t msg, void *data)
                                                         ^~~~
../ucglib/csrc/ucg_dev_default_cb.c: In function 'ucg_handle_l90se':
../ucglib/csrc/ucg_dev_default_cb.c:220:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i, j;
               ^
../ucglib/csrc/ucg_dev_default_cb.c:206:11: note: shadowed declaration is here
   uint8_t i;
           ^
  CC	../ucglib/csrc/ucg_dev_oled_160x128_samsung.c
  CC	../ucglib/csrc/ucg_dev_ic_seps225.c
../ucglib/csrc/ucg_dev_ic_seps225.c:114:11: warning: no previous prototype for 'ucg_handle_seps225_l90fx' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_seps225_l90fx(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_seps225.c:168:11: warning: no previous prototype for 'ucg_handle_seps225_l90se' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_seps225_l90se(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_seps225.c: In function 'ucg_handle_seps225_l90se':
../ucglib/csrc/ucg_dev_ic_seps225.c:197:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i, j;
               ^
../ucglib/csrc/ucg_dev_ic_seps225.c:170:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_seps225.c: In function 'ucg_ext_seps225_16':
../ucglib/csrc/ucg_dev_ic_seps225.c:311:63: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_seps225_16(ucg_t *ucg, ucg_int_t msg, void *data)
                                                               ^~~~
  CC	../ucglib/csrc/ucg_ccs.c
  CC	../ucglib/csrc/ucg_dev_tft_240x320_itdb02.c
  CC	../ucglib/csrc/ucg_dev_ic_ili9486.c
../ucglib/csrc/ucg_dev_ic_ili9486.c:114:11: warning: no previous prototype for 'ucg_handle_ili9486_l90fx' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ili9486_l90fx(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ili9486.c:158:11: warning: no previous prototype for 'ucg_handle_ili9486_l90tc' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ili9486_l90tc(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ili9486.c:250:11: warning: no previous prototype for 'ucg_handle_ili9486_l90se' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ili9486_l90se(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ili9486.c: In function 'ucg_handle_ili9486_l90se':
../ucglib/csrc/ucg_dev_ic_ili9486.c:267:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i;
               ^
../ucglib/csrc/ucg_dev_ic_ili9486.c:252:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_ili9486.c: In function 'ucg_ext_ili9486_18':
../ucglib/csrc/ucg_dev_ic_ili9486.c:359:63: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_ili9486_18(ucg_t *ucg, ucg_int_t msg, void *data)
                                                               ^~~~
  CC	../ucglib/csrc/ucg_dev_ic_ssd1331.c
../ucglib/csrc/ucg_dev_ic_ssd1331.c:63:11: warning: no previous prototype for 'ucg_handle_ssd1331_l90fx' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ssd1331_l90fx(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ssd1331.c:122:11: warning: no previous prototype for 'ucg_handle_ssd1331_l90tc' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ssd1331_l90tc(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ssd1331.c:231:11: warning: no previous prototype for 'ucg_handle_ssd1331_l90se' [-Wmissing-prototypes]
 ucg_int_t ucg_handle_ssd1331_l90se(ucg_t *ucg)
           ^~~~~~~~~~~~~~~~~~~~~~~~
../ucglib/csrc/ucg_dev_ic_ssd1331.c: In function 'ucg_handle_ssd1331_l90se':
../ucglib/csrc/ucg_dev_ic_ssd1331.c:260:15: warning: declaration of 'i' shadows a previous local [-Wshadow]
     ucg_int_t i, j;
               ^
../ucglib/csrc/ucg_dev_ic_ssd1331.c:233:11: note: shadowed declaration is here
   uint8_t i;
           ^
../ucglib/csrc/ucg_dev_ic_ssd1331.c: In function 'ucg_ext_ssd1331_18':
../ucglib/csrc/ucg_dev_ic_ssd1331.c:362:63: warning: unused parameter 'data' [-Wunused-parameter]
 ucg_int_t ucg_ext_ssd1331_18(ucg_t *ucg, ucg_int_t msg, void *data)
                                                               ^~~~
```

--- a/csrc/ucg_circle.c
+++ b/csrc/ucg_circle.c
@@ -70,7 +70,9 @@ static void ucg_draw_circle_section(ucg_t *ucg, ucg_int_t x, ucg_int_t y, ucg_in
     }
 }
 
-void ucg_draw_circle(ucg_t *ucg, ucg_int_t x0, ucg_int_t y0, ucg_int_t rad, uint8_t option)
+static void ucg_draw_circle(ucg_t *ucg, ucg_int_t x0, ucg_int_t y0, ucg_int_t rad, uint8_t option);
+
+static void ucg_draw_circle(ucg_t *ucg, ucg_int_t x0, ucg_int_t y0, ucg_int_t rad, uint8_t option)
 {
     ucg_int_t f;
     ucg_int_t ddF_x;
@@ -159,7 +161,9 @@ static void ucg_draw_disc_section(ucg_t *ucg, ucg_int_t x, ucg_int_t y, ucg_int_
     }
 }
 
-void ucg_draw_disc(ucg_t *ucg, ucg_int_t x0, ucg_int_t y0, ucg_int_t rad, uint8_t option)
+static void ucg_draw_disc(ucg_t *ucg, ucg_int_t x0, ucg_int_t y0, ucg_int_t rad, uint8_t option);
+
+static void ucg_draw_disc(ucg_t *ucg, ucg_int_t x0, ucg_int_t y0, ucg_int_t rad, uint8_t option)
 {
   ucg_int_t f;
   ucg_int_t ddF_x;

--- a/csrc/ucg_com_msg_api.c
+++ b/csrc/ucg_com_msg_api.c
@@ -35,8 +35,15 @@
 
 #include "ucg.h"
 
-int16_t ucg_com_template_cb(ucg_t *ucg, int16_t msg, uint32_t arg, uint8_t *data)
+static int16_t ucg_com_template_cb(ucg_t *ucg, int16_t msg, uint32_t arg, uint8_t *data);
+static void ucg_com_SendStringP(ucg_t *ucg, uint16_t cnt, const ucg_pgm_uint8_t *byte_ptr);
+
+static __attribute__((unused)) int16_t ucg_com_template_cb(ucg_t *ucg, int16_t msg, uint32_t arg, uint8_t *data)
 {
+  (void)ucg;
+  (void)arg;
+  (void)data;
+
   switch(msg)
   {
     case UCG_COM_MSG_POWER_UP:
@@ -179,7 +186,7 @@ void ucg_com_SendString(ucg_t *ucg, uint16_t cnt, const uint8_t *byte_ptr)
   ucg->com_cb(ucg, UCG_COM_MSG_SEND_STR, cnt, (uint8_t *)byte_ptr);
 }
 
-void ucg_com_SendStringP(ucg_t *ucg, uint16_t cnt, const ucg_pgm_uint8_t *byte_ptr)
+static void ucg_com_SendStringP(ucg_t *ucg, uint16_t cnt, const ucg_pgm_uint8_t *byte_ptr)
 {
   uint8_t b;
   while( cnt > 0 )

--- a/csrc/ucg_dev_default_cb.c
+++ b/csrc/ucg_dev_default_cb.c
@@ -59,6 +59,10 @@ ucg_int_t ucg_dev_default_cb(ucg_t *ucg, ucg_int_t msg, void *data)
 */
 ucg_int_t ucg_ext_none(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)ucg;
+  (void)msg;
+  (void)data;
+
   return 1;	/* all ok */  
 }
 
@@ -217,7 +221,7 @@ ucg_int_t ucg_handle_l90se(ucg_t *ucg, ucg_dev_fnptr dev_cb)
   if ( ucg_clip_l90se(ucg) != 0 )
   {
     ucg_int_t dx, dy;
-    ucg_int_t i, j;
+    ucg_int_t k, j;
     switch(ucg->arg.dir)
     {
       case 0: dx = 1; dy = 0; break;
@@ -226,7 +230,7 @@ ucg_int_t ucg_handle_l90se(ucg_t *ucg, ucg_dev_fnptr dev_cb)
       case 3: dx = 0; dy = -1; break;
       default: dx = 1; dy = 0; break;	/* avoid compiler warning */
     }
-    for( i = 0; i < ucg->arg.len; i++ )
+    for( k = 0; k < ucg->arg.len; k++ )
     {
       ucg->arg.pixel.rgb.color[0] = ucg->arg.ccs_line[0].current;
       ucg->arg.pixel.rgb.color[1] = ucg->arg.ccs_line[1].current; 

--- a/csrc/ucg_dev_ic_hx8352c.c
+++ b/csrc/ucg_dev_ic_hx8352c.c
@@ -37,6 +37,12 @@
 
 #include "ucg.h"
 
+static ucg_int_t ucg_handle_hx8352c_l90fx(ucg_t *ucg);
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_hx8352c_l90tc(ucg_t *ucg);
+#endif
+static ucg_int_t ucg_handle_hx8352c_l90se(ucg_t *ucg);
+
 const ucg_pgm_uint8_t ucg_hx8352c_set_pos_seq[] =
     {
         UCG_CS(0), /* enable chip */
@@ -137,7 +143,7 @@ const ucg_pgm_uint8_t ucg_hx8352c_set_pos_dir3_seq[] =
         UCG_DATA(),     /* change to data mode */
         UCG_END()};
 
-ucg_int_t ucg_handle_hx8352c_l90fx(ucg_t *ucg)
+static ucg_int_t ucg_handle_hx8352c_l90fx(ucg_t *ucg)
 {
   uint8_t c[3];
   ucg_int_t tmp;
@@ -177,7 +183,8 @@ ucg_int_t ucg_handle_hx8352c_l90fx(ucg_t *ucg)
 */
 
 /* with CmdDataSequence */
-ucg_int_t ucg_handle_hx8352c_l90tc(ucg_t *ucg)
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_hx8352c_l90tc(ucg_t *ucg)
 {
   if (ucg_clip_l90tc(ucg) != 0)
   {
@@ -274,8 +281,9 @@ ucg_int_t ucg_handle_hx8352c_l90tc(ucg_t *ucg)
   }
   return 0;
 }
+#endif
 
-ucg_int_t ucg_handle_hx8352c_l90se(ucg_t *ucg)
+static ucg_int_t ucg_handle_hx8352c_l90se(ucg_t *ucg)
 {
   uint8_t i;
   uint8_t c[3];
@@ -303,7 +311,7 @@ ucg_int_t ucg_handle_hx8352c_l90se(ucg_t *ucg)
 
   if (ucg_clip_l90se(ucg) != 0)
   {
-    ucg_int_t i, j;
+    ucg_int_t k;
     switch (ucg->arg.dir)
     {
     case 0:
@@ -327,8 +335,7 @@ ucg_int_t ucg_handle_hx8352c_l90se(ucg_t *ucg)
       break;
     }
 
-    j = 0;
-    for (i = 0; i < ucg->arg.len; i++)
+    for (k = 0; k < ucg->arg.len; k++)
     {
       c[0] = ucg->arg.ccs_line[0].current;
       c[1] = ucg->arg.ccs_line[1].current;
@@ -403,6 +410,8 @@ ucg_int_t ucg_dev_ic_hx8352c_18(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_hx8352c_18(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch (msg)
   {
   case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_dev_ic_ili9163.c
+++ b/csrc/ucg_dev_ic_ili9163.c
@@ -39,6 +39,11 @@ Code is for 128x128 display which is shifted by 32 pixel within the controller R
 
 #include "ucg.h"
 
+static ucg_int_t ucg_handle_ili9163_l90fx(ucg_t *ucg);
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_ili9163_l90tc(ucg_t *ucg);
+#endif
+static ucg_int_t ucg_handle_ili9163_l90se(ucg_t *ucg);
 
 const ucg_pgm_uint8_t ucg_ili9163_set_pos_seq[] = 
 {
@@ -122,7 +127,7 @@ const ucg_pgm_uint8_t ucg_ili9163_set_pos_dir3_seq[] =
   UCG_END()
 };
 
-ucg_int_t ucg_handle_ili9163_l90fx(ucg_t *ucg)
+static ucg_int_t ucg_handle_ili9163_l90fx(ucg_t *ucg)
 {
   uint8_t c[3];
   ucg_int_t tmp;
@@ -172,8 +177,9 @@ ucg_int_t ucg_handle_ili9163_l90fx(ucg_t *ucg)
   
 */
 
+#ifdef UCG_MSG_DRAW_L90TC
 /* with CmdDataSequence */ 
-ucg_int_t ucg_handle_ili9163_l90tc(ucg_t *ucg)
+static ucg_int_t ucg_handle_ili9163_l90tc(ucg_t *ucg)
 {
   if ( ucg_clip_l90tc(ucg) != 0 )
   {
@@ -262,9 +268,9 @@ ucg_int_t ucg_handle_ili9163_l90tc(ucg_t *ucg)
   }
   return 0;
 }
+#endif
 
-
-ucg_int_t ucg_handle_ili9163_l90se(ucg_t *ucg)
+static ucg_int_t ucg_handle_ili9163_l90se(ucg_t *ucg)
 {
   uint8_t i;
   uint8_t c[3];
@@ -281,7 +287,7 @@ ucg_int_t ucg_handle_ili9163_l90se(ucg_t *ucg)
   
   if ( ucg_clip_l90se(ucg) != 0 )
   {
-    ucg_int_t i;
+    ucg_int_t k;
     switch(ucg->arg.dir)
     {
       case 0: 
@@ -311,7 +317,7 @@ ucg_int_t ucg_handle_ili9163_l90se(ucg_t *ucg)
 	break;
     }
     
-    for( i = 0; i < ucg->arg.len; i++ )
+    for( k = 0; k < ucg->arg.len; k++ )
     {
       c[0] = ucg->arg.ccs_line[0].current;
       c[1] = ucg->arg.ccs_line[1].current; 
@@ -391,6 +397,8 @@ ucg_int_t ucg_dev_ic_ili9163_18(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_ili9163_18(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch(msg)
   {
     case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_dev_ic_ili9325.c
+++ b/csrc/ucg_dev_ic_ili9325.c
@@ -37,6 +37,7 @@
 
 #include "ucg.h"
 
+static ucg_int_t xxxxxx_ucg_handle_ili9325_l90tc(ucg_t *ucg);
 
 static const ucg_pgm_uint8_t ucg_ili9325_set_pos_seq[] = 
 {
@@ -172,7 +173,7 @@ static const ucg_pgm_uint8_t ucg_ili9325_set_y_pos_seq[] =
 };
 
 /* without CmdDataSequence */ 
-ucg_int_t xxxxxx_ucg_handle_ili9325_l90tc(ucg_t *ucg)
+static __attribute__((unused)) ucg_int_t xxxxxx_ucg_handle_ili9325_l90tc(ucg_t *ucg)
 {
   if ( ucg_clip_l90tc(ucg) != 0 )
   {
@@ -236,7 +237,7 @@ ucg_int_t xxxxxx_ucg_handle_ili9325_l90tc(ucg_t *ucg)
   return 0;
 }
 
-
+#ifdef UCG_MSG_DRAW_L90TC
 /* with CmdDataSequence */ 
 static ucg_int_t ucg_handle_ili9325_l90tc(ucg_t *ucg)
 {
@@ -366,6 +367,7 @@ static ucg_int_t ucg_handle_ili9325_l90tc(ucg_t *ucg)
   }
   return 0;
 }
+#endif
 
 
 static ucg_int_t ucg_handle_ili9325_l90se(ucg_t *ucg)
@@ -384,7 +386,7 @@ static ucg_int_t ucg_handle_ili9325_l90se(ucg_t *ucg)
   
   if ( ucg_clip_l90se(ucg) != 0 )
   {
-    ucg_int_t i;
+    ucg_int_t k;
     switch(ucg->arg.dir)
     {
       case 0: 
@@ -402,7 +404,7 @@ static ucg_int_t ucg_handle_ili9325_l90se(ucg_t *ucg)
 	break;
     }
     
-    for( i = 0; i < ucg->arg.len; i++ )
+    for( k = 0; k < ucg->arg.len; k++ )
     {
       c[0] = ucg->arg.ccs_line[0].current;
       c[1] = ucg->arg.ccs_line[1].current; 
@@ -473,6 +475,8 @@ ucg_int_t ucg_dev_ic_ili9325_18(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_ili9325_18(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch(msg)
   {
     case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_dev_ic_ili9325_spi.c
+++ b/csrc/ucg_dev_ic_ili9325_spi.c
@@ -173,7 +173,7 @@ static const ucg_pgm_uint8_t ucg_ili9325_set_y_pos_seq[] =
 };
 
 /* without CmdDataSequence */ 
-static ucg_int_t xxxxxx_ucg_handle_ili9325_l90tc(ucg_t *ucg)
+static __attribute__((unused)) ucg_int_t xxxxxx_ucg_handle_ili9325_l90tc(ucg_t *ucg)
 {
   if ( ucg_clip_l90tc(ucg) != 0 )
   {
@@ -237,7 +237,7 @@ static ucg_int_t xxxxxx_ucg_handle_ili9325_l90tc(ucg_t *ucg)
   return 0;
 }
 
-
+#ifdef UCG_MSG_DRAW_L90TC
 /* with CmdDataSequence */ 
 static ucg_int_t ucg_handle_ili9325_l90tc(ucg_t *ucg)
 {
@@ -367,7 +367,7 @@ static ucg_int_t ucg_handle_ili9325_l90tc(ucg_t *ucg)
   }
   return 0;
 }
-
+#endif
 
 static ucg_int_t ucg_handle_ili9325_l90se(ucg_t *ucg)
 {
@@ -385,7 +385,7 @@ static ucg_int_t ucg_handle_ili9325_l90se(ucg_t *ucg)
   
   if ( ucg_clip_l90se(ucg) != 0 )
   {
-    ucg_int_t i;
+    ucg_int_t k;
     switch(ucg->arg.dir)
     {
       case 0: 
@@ -403,7 +403,7 @@ static ucg_int_t ucg_handle_ili9325_l90se(ucg_t *ucg)
 	break;
     }
     
-    for( i = 0; i < ucg->arg.len; i++ )
+    for( k = 0; k < ucg->arg.len; k++ )
     {
       c[0] = ucg->arg.ccs_line[0].current;
       c[1] = ucg->arg.ccs_line[1].current; 
@@ -473,6 +473,8 @@ ucg_int_t ucg_dev_ic_ili9325_spi_18(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_ili9325_spi_18(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch(msg)
   {
     case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_dev_ic_ili9341.c
+++ b/csrc/ucg_dev_ic_ili9341.c
@@ -37,6 +37,11 @@
 
 #include "ucg.h"
 
+static ucg_int_t ucg_handle_ili9341_l90fx(ucg_t *ucg);
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_ili9341_l90tc(ucg_t *ucg);
+#endif
+static ucg_int_t ucg_handle_ili9341_l90se(ucg_t *ucg);
 
 const ucg_pgm_uint8_t ucg_ili9341_set_pos_seq[] = 
 {
@@ -120,7 +125,7 @@ const ucg_pgm_uint8_t ucg_ili9341_set_pos_dir3_seq[] =
   UCG_END()
 };
 
-ucg_int_t ucg_handle_ili9341_l90fx(ucg_t *ucg)
+static ucg_int_t ucg_handle_ili9341_l90fx(ucg_t *ucg)
 {
   uint8_t c[3];
   ucg_int_t tmp;
@@ -164,7 +169,8 @@ ucg_int_t ucg_handle_ili9341_l90fx(ucg_t *ucg)
 */
 
 /* with CmdDataSequence */ 
-ucg_int_t ucg_handle_ili9341_l90tc(ucg_t *ucg)
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_ili9341_l90tc(ucg_t *ucg)
 {
   if ( ucg_clip_l90tc(ucg) != 0 )
   {
@@ -250,9 +256,9 @@ ucg_int_t ucg_handle_ili9341_l90tc(ucg_t *ucg)
   }
   return 0;
 }
+#endif
 
-
-ucg_int_t ucg_handle_ili9341_l90se(ucg_t *ucg)
+static ucg_int_t ucg_handle_ili9341_l90se(ucg_t *ucg)
 {
   uint8_t i;
   uint8_t c[3];
@@ -269,7 +275,7 @@ ucg_int_t ucg_handle_ili9341_l90se(ucg_t *ucg)
   
   if ( ucg_clip_l90se(ucg) != 0 )
   {
-    ucg_int_t i;
+    ucg_int_t k;
     switch(ucg->arg.dir)
     {
       case 0: 
@@ -293,7 +299,7 @@ ucg_int_t ucg_handle_ili9341_l90se(ucg_t *ucg)
 	break;
     }
     
-    for( i = 0; i < ucg->arg.len; i++ )
+    for( k = 0; k < ucg->arg.len; k++ )
     {
       c[0] = ucg->arg.ccs_line[0].current;
       c[1] = ucg->arg.ccs_line[1].current; 
@@ -371,6 +377,8 @@ ucg_int_t ucg_dev_ic_ili9341_18(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_ili9341_18(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch(msg)
   {
     case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_dev_ic_ili9486.c
+++ b/csrc/ucg_dev_ic_ili9486.c
@@ -37,6 +37,12 @@
 
 #include "ucg.h"
 
+static ucg_int_t ucg_handle_ili9486_l90fx(ucg_t *ucg);
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_ili9486_l90tc(ucg_t *ucg);
+#endif
+static ucg_int_t ucg_handle_ili9486_l90se(ucg_t *ucg);
+
 const ucg_pgm_uint8_t ucg_ili9486_set_pos_seq[] =
     {
         UCG_CS(0), /* enable chip */
@@ -111,7 +117,7 @@ const ucg_pgm_uint8_t ucg_ili9486_set_pos_dir3_seq[] =
         UCG_DATA(),     /* change to data mode */
         UCG_END()};
 
-ucg_int_t ucg_handle_ili9486_l90fx(ucg_t *ucg)
+static ucg_int_t ucg_handle_ili9486_l90fx(ucg_t *ucg)
 {
   uint8_t c[3];
   ucg_int_t tmp;
@@ -154,8 +160,9 @@ ucg_int_t ucg_handle_ili9486_l90fx(ucg_t *ucg)
   
 */
 
+#ifdef UCG_MSG_DRAW_L90TC
 /* with CmdDataSequence */
-ucg_int_t ucg_handle_ili9486_l90tc(ucg_t *ucg)
+static ucg_int_t ucg_handle_ili9486_l90tc(ucg_t *ucg)
 {
   if (ucg_clip_l90tc(ucg) != 0)
   {
@@ -246,8 +253,9 @@ ucg_int_t ucg_handle_ili9486_l90tc(ucg_t *ucg)
   }
   return 0;
 }
+#endif
 
-ucg_int_t ucg_handle_ili9486_l90se(ucg_t *ucg)
+static ucg_int_t ucg_handle_ili9486_l90se(ucg_t *ucg)
 {
   uint8_t i;
   uint8_t c[3];
@@ -264,7 +272,7 @@ ucg_int_t ucg_handle_ili9486_l90se(ucg_t *ucg)
 
   if (ucg_clip_l90se(ucg) != 0)
   {
-    ucg_int_t i;
+    ucg_int_t k;
     switch (ucg->arg.dir)
     {
     case 0:
@@ -288,7 +296,7 @@ ucg_int_t ucg_handle_ili9486_l90se(ucg_t *ucg)
       break;
     }
 
-    for (i = 0; i < ucg->arg.len; i++)
+    for (k = 0; k < ucg->arg.len; k++)
     {
       c[0] = ucg->arg.ccs_line[0].current;
       c[1] = ucg->arg.ccs_line[1].current;
@@ -358,6 +366,8 @@ ucg_int_t ucg_dev_ic_ili9486_18(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_ili9486_18(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch (msg)
   {
   case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_dev_ic_ld50t6160.c
+++ b/csrc/ucg_dev_ic_ld50t6160.c
@@ -37,6 +37,8 @@
 
 #include "ucg.h"
 
+static ucg_int_t ucg_handle_ld50t6160_l90se(ucg_t *ucg);
+
 /*
   Graphics Memory Writing Direction: 0x05
   Bit 3: RGB/BGR
@@ -223,7 +225,7 @@ static ucg_int_t ucg_handle_ld50t6160_l90tc(ucg_t *ucg)
 }
 #endif /* UCG_MSG_DRAW_L90TC */
 
-ucg_int_t ucg_handle_ld50t6160_l90se(ucg_t *ucg)
+static ucg_int_t ucg_handle_ld50t6160_l90se(ucg_t *ucg)
 {
   uint8_t i;
   uint8_t c[3];
@@ -240,7 +242,7 @@ ucg_int_t ucg_handle_ld50t6160_l90se(ucg_t *ucg)
   
   if ( ucg_clip_l90se(ucg) != 0 )
   {
-    ucg_int_t i;
+    ucg_int_t k;
     switch(ucg->arg.dir)
     {
       case 0: 
@@ -264,7 +266,7 @@ ucg_int_t ucg_handle_ld50t6160_l90se(ucg_t *ucg)
 	break;
     }
     
-    for( i = 0; i < ucg->arg.len; i++ )
+    for( k = 0; k < ucg->arg.len; k++ )
     {
       c[0] = ucg->arg.ccs_line[0].current>>2;
       c[1] = ucg->arg.ccs_line[1].current>>2; 
@@ -347,6 +349,8 @@ ucg_int_t ucg_dev_ic_ld50t6160_18(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_ld50t6160_18(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch(msg)
   {
     case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_dev_ic_pcf8833.c
+++ b/csrc/ucg_dev_ic_pcf8833.c
@@ -42,6 +42,12 @@
 #define WIDTH 132
 #define HEIGHT 132
 
+static ucg_int_t ucg_handle_pcf8833_l90fx(ucg_t *ucg);
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_pcf8833_l90tc(ucg_t *ucg);
+#endif
+static ucg_int_t ucg_handle_pcf8833_l90se(ucg_t *ucg);
+
 const ucg_pgm_uint8_t ucg_pcf8833_set_pos_seq[] = 
 {
   UCG_CS(0),					/* enable chip */
@@ -146,7 +152,7 @@ static uint8_t ucg_pcf8833_get_color_low_byte(ucg_t *ucg)
     return ((((ucg->arg.pixel.rgb.color[1]))<<3)&0x0e0) | (((ucg->arg.pixel.rgb.color[2]) >>3));
 }
 
-ucg_int_t ucg_handle_pcf8833_l90fx(ucg_t *ucg)
+static ucg_int_t ucg_handle_pcf8833_l90fx(ucg_t *ucg)
 {
   uint8_t c[3];
   ucg_int_t tmp;
@@ -188,8 +194,9 @@ ucg_int_t ucg_handle_pcf8833_l90fx(ucg_t *ucg)
   
 */
 
+#ifdef UCG_MSG_DRAW_L90TC
 /* with CmdDataSequence */ 
-ucg_int_t ucg_handle_pcf8833_l90tc(ucg_t *ucg)
+static ucg_int_t ucg_handle_pcf8833_l90tc(ucg_t *ucg)
 {
   if ( ucg_clip_l90tc(ucg) != 0 )
   {
@@ -272,9 +279,9 @@ ucg_int_t ucg_handle_pcf8833_l90tc(ucg_t *ucg)
   }
   return 0;
 }
+#endif
 
-
-ucg_int_t ucg_handle_pcf8833_l90se(ucg_t *ucg)
+static ucg_int_t ucg_handle_pcf8833_l90se(ucg_t *ucg)
 {
   uint8_t i;
   uint8_t c[3];
@@ -291,7 +298,7 @@ ucg_int_t ucg_handle_pcf8833_l90se(ucg_t *ucg)
   
   if ( ucg_clip_l90se(ucg) != 0 )
   {
-    ucg_int_t i;
+    ucg_int_t k;
     switch(ucg->arg.dir)
     {
       case 0: 
@@ -315,7 +322,7 @@ ucg_int_t ucg_handle_pcf8833_l90se(ucg_t *ucg)
 	break;
     }
     
-    for( i = 0; i < ucg->arg.len; i++ )
+    for( k = 0; k < ucg->arg.len; k++ )
     {
       
       c[0] = (ucg->arg.ccs_line[0].current&0x0f8) | (((ucg->arg.ccs_line[1].current) >>5));
@@ -398,6 +405,8 @@ ucg_int_t ucg_dev_ic_pcf8833_16(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_pcf8833_16(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch(msg)
   {
     case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_dev_ic_seps225.c
+++ b/csrc/ucg_dev_ic_seps225.c
@@ -39,6 +39,9 @@
 
 #include "ucg.h"
 
+static ucg_int_t ucg_handle_seps225_l90fx(ucg_t *ucg);
+static ucg_int_t ucg_handle_seps225_l90se(ucg_t *ucg);
+
 static uint8_t ucg_seps225_get_color_high_byte(ucg_t *ucg)
 {
     return (ucg->arg.pixel.rgb.color[0]&0x0f8) | (((ucg->arg.pixel.rgb.color[1]) >>5));
@@ -111,7 +114,7 @@ static const ucg_pgm_uint8_t ucg_seps255_pos_dir3_seq[] =
 };
 #endif 
 
-ucg_int_t ucg_handle_seps225_l90fx(ucg_t *ucg)
+static ucg_int_t ucg_handle_seps225_l90fx(ucg_t *ucg)
 {
   if ( ucg_clip_l90fx(ucg) != 0 )
   {
@@ -165,7 +168,7 @@ ucg_int_t ucg_handle_seps225_l90fx(ucg_t *ucg)
 */
 
 
-ucg_int_t ucg_handle_seps225_l90se(ucg_t *ucg)
+static ucg_int_t ucg_handle_seps225_l90se(ucg_t *ucg)
 {
   uint8_t i;
   uint8_t c[3];
@@ -194,7 +197,7 @@ ucg_int_t ucg_handle_seps225_l90se(ucg_t *ucg)
   if ( ucg_clip_l90se(ucg) != 0 )
   {
     ucg_int_t dx, dy;
-    ucg_int_t i, j;
+    ucg_int_t k, j;
     //uint8_t r, g, b;
     switch(ucg->arg.dir)
     {
@@ -216,7 +219,7 @@ ucg_int_t ucg_handle_seps225_l90se(ucg_t *ucg)
 	break;
       default: dx = 1; dy = 0; break;	/* avoid compiler warning */
     }
-    for( i = 0; i < ucg->arg.len; i++ )
+    for( k = 0; k < ucg->arg.len; k++ )
     {
       /*
       r = ucg->arg.ccs_line[0].current;
@@ -310,6 +313,8 @@ ucg_int_t ucg_dev_ic_seps225_16(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_seps225_16(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch(msg)
   {
     case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_dev_ic_seps225.c
+++ b/csrc/ucg_dev_ic_seps225.c
@@ -294,12 +294,12 @@ ucg_int_t ucg_dev_ic_seps225_16(ucg_t *ucg, ucg_int_t msg, void *data)
       return 1;
 #ifdef UCG_MSG_DRAW_L90TC
     case UCG_MSG_DRAW_L90TC:
-      ucg_handle_l90tc(ucg, ucg_dev_ic_seps225_18);
+      ucg_handle_l90tc(ucg, ucg_dev_ic_seps225_16);
       return 1;
 #endif /* UCG_MSG_DRAW_L90TC */
 #ifdef UCG_MSG_DRAW_L90BF
      case UCG_MSG_DRAW_L90BF:
-      ucg_handle_l90bf(ucg, ucg_dev_ic_seps225_18);
+      ucg_handle_l90bf(ucg, ucg_dev_ic_seps225_16);
       return 1;
 #endif /* UCG_MSG_DRAW_L90BF */
     /* msg UCG_MSG_DRAW_L90SE is handled by ucg_dev_default_cb */

--- a/csrc/ucg_dev_ic_ssd1331.c
+++ b/csrc/ucg_dev_ic_ssd1331.c
@@ -38,9 +38,11 @@
 
 #include "ucg.h"
 
-
-//ucg_int_t u8g_dev_ic_ssd1331(ucg_t *ucg, ucg_int_t msg, void *data);
-
+static ucg_int_t ucg_handle_ssd1331_l90fx(ucg_t *ucg);
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_ssd1331_l90tc(ucg_t *ucg);
+#endif
+static ucg_int_t ucg_handle_ssd1331_l90se(ucg_t *ucg);
 
 const ucg_pgm_uint8_t ucg_ssd1331_set_pos_dir0_seq[] = 
 {
@@ -60,7 +62,7 @@ const ucg_pgm_uint8_t ucg_ssd1331_set_pos_dir1_seq[] =
   UCG_END()
 };
 
-ucg_int_t ucg_handle_ssd1331_l90fx(ucg_t *ucg)
+static ucg_int_t ucg_handle_ssd1331_l90fx(ucg_t *ucg)
 {
   uint8_t c[3];
   if ( ucg_clip_l90fx(ucg) != 0 )
@@ -118,8 +120,8 @@ const ucg_pgm_uint8_t ucg_ssd1331_set_pos_for_y_seq[] =
   UCG_END()
 };
 
-
-ucg_int_t ucg_handle_ssd1331_l90tc(ucg_t *ucg)
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_ssd1331_l90tc(ucg_t *ucg)
 {
   if ( ucg_clip_l90tc(ucg) != 0 )
   {
@@ -226,9 +228,9 @@ ucg_int_t ucg_handle_ssd1331_l90tc(ucg_t *ucg)
   }
   return 0;
 }
+#endif
 
-
-ucg_int_t ucg_handle_ssd1331_l90se(ucg_t *ucg)
+static ucg_int_t ucg_handle_ssd1331_l90se(ucg_t *ucg)
 {
   uint8_t i;
   uint8_t c[3];
@@ -257,7 +259,7 @@ ucg_int_t ucg_handle_ssd1331_l90se(ucg_t *ucg)
   if ( ucg_clip_l90se(ucg) != 0 )
   {
     ucg_int_t dx, dy;
-    ucg_int_t i, j;
+    ucg_int_t k, j;
     switch(ucg->arg.dir)
     {
       case 0: dx = 1; dy = 0; 
@@ -278,7 +280,7 @@ ucg_int_t ucg_handle_ssd1331_l90se(ucg_t *ucg)
 	break;
       default: dx = 1; dy = 0; break;	/* avoid compiler warning */
     }
-    for( i = 0; i < ucg->arg.len; i++ )
+    for( k = 0; k < ucg->arg.len; k++ )
     {
       c[0] = ucg->arg.ccs_line[0].current >> 2;
       c[1] = ucg->arg.ccs_line[1].current >> 2; 
@@ -361,6 +363,8 @@ ucg_int_t ucg_dev_ic_ssd1331_18(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_ssd1331_18(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch(msg)
   {
     case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_dev_ic_ssd1351.c
+++ b/csrc/ucg_dev_ic_ssd1351.c
@@ -37,9 +37,11 @@
 
 #include "ucg.h"
 
-
-//ucg_int_t u8g_dev_ic_ssd1351(ucg_t *ucg, ucg_int_t msg, void *data);
-
+static ucg_int_t ucg_handle_ssd1351_l90fx(ucg_t *ucg);
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_ssd1351_l90tc(ucg_t *ucg);
+#endif
+static ucg_int_t ucg_handle_ssd1351_l90se(ucg_t *ucg);
 
 const ucg_pgm_uint8_t ucg_ssd1351_set_pos_dir0_seq[] = 
 {
@@ -61,7 +63,7 @@ const ucg_pgm_uint8_t ucg_ssd1351_set_pos_dir1_seq[] =
   UCG_END()
 };
 
-ucg_int_t ucg_handle_ssd1351_l90fx(ucg_t *ucg)
+static ucg_int_t ucg_handle_ssd1351_l90fx(ucg_t *ucg)
 {
   uint8_t c[3];
   if ( ucg_clip_l90fx(ucg) != 0 )
@@ -119,8 +121,8 @@ const ucg_pgm_uint8_t ucg_ssd1351_set_pos_for_y_seq[] =
   UCG_END()
 };
 
-
-ucg_int_t ucg_handle_ssd1351_l90tc(ucg_t *ucg)
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_ssd1351_l90tc(ucg_t *ucg)
 {
   if ( ucg_clip_l90tc(ucg) != 0 )
   {
@@ -227,7 +229,7 @@ ucg_int_t ucg_handle_ssd1351_l90tc(ucg_t *ucg)
   }
   return 0;
 }
-
+#endif
 
 ucg_int_t ucg_handle_ssd1351_l90se(ucg_t *ucg)
 {
@@ -258,7 +260,7 @@ ucg_int_t ucg_handle_ssd1351_l90se(ucg_t *ucg)
   if ( ucg_clip_l90se(ucg) != 0 )
   {
     ucg_int_t dx, dy;
-    ucg_int_t i, j;
+    ucg_int_t k, j;
     switch(ucg->arg.dir)
     {
       case 0: dx = 1; dy = 0; 
@@ -279,7 +281,7 @@ ucg_int_t ucg_handle_ssd1351_l90se(ucg_t *ucg)
 	break;
       default: dx = 1; dy = 0; break;	/* avoid compiler warning */
     }
-    for( i = 0; i < ucg->arg.len; i++ )
+    for( k = 0; k < ucg->arg.len; k++ )
     {
       c[0] = ucg->arg.ccs_line[0].current >> 2;
       c[1] = ucg->arg.ccs_line[1].current >> 2; 
@@ -361,6 +363,8 @@ ucg_int_t ucg_dev_ic_ssd1351_18(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_ssd1351_18(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch(msg)
   {
     case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_dev_ic_st7735.c
+++ b/csrc/ucg_dev_ic_st7735.c
@@ -37,6 +37,11 @@
 
 #include "ucg.h"
 
+static ucg_int_t ucg_handle_st7735_l90fx(ucg_t *ucg);
+#ifdef UCG_MSG_DRAW_L90TC
+static ucg_int_t ucg_handle_st7735_l90tc(ucg_t *ucg);
+#endif
+static ucg_int_t ucg_handle_st7735_l90se(ucg_t *ucg);
 
 const ucg_pgm_uint8_t ucg_st7735_set_pos_seq[] = 
 {
@@ -120,7 +125,7 @@ const ucg_pgm_uint8_t ucg_st7735_set_pos_dir3_seq[] =
   UCG_END()
 };
 
-ucg_int_t ucg_handle_st7735_l90fx(ucg_t *ucg)
+static ucg_int_t ucg_handle_st7735_l90fx(ucg_t *ucg)
 {
   uint8_t c[3];
   ucg_int_t tmp;
@@ -163,8 +168,9 @@ ucg_int_t ucg_handle_st7735_l90fx(ucg_t *ucg)
   
 */
 
+#ifdef UCG_MSG_DRAW_L90TC
 /* with CmdDataSequence */ 
-ucg_int_t ucg_handle_st7735_l90tc(ucg_t *ucg)
+static ucg_int_t ucg_handle_st7735_l90tc(ucg_t *ucg)
 {
   if ( ucg_clip_l90tc(ucg) != 0 )
   {
@@ -250,9 +256,9 @@ ucg_int_t ucg_handle_st7735_l90tc(ucg_t *ucg)
   }
   return 0;
 }
+#endif
 
-
-ucg_int_t ucg_handle_st7735_l90se(ucg_t *ucg)
+static ucg_int_t ucg_handle_st7735_l90se(ucg_t *ucg)
 {
   uint8_t i;
   uint8_t c[3];
@@ -269,7 +275,7 @@ ucg_int_t ucg_handle_st7735_l90se(ucg_t *ucg)
   
   if ( ucg_clip_l90se(ucg) != 0 )
   {
-    ucg_int_t i;
+    ucg_int_t k;
     switch(ucg->arg.dir)
     {
       case 0: 
@@ -293,7 +299,7 @@ ucg_int_t ucg_handle_st7735_l90se(ucg_t *ucg)
 	break;
     }
     
-    for( i = 0; i < ucg->arg.len; i++ )
+    for( k = 0; k < ucg->arg.len; k++ )
     {
       c[0] = ucg->arg.ccs_line[0].current;
       c[1] = ucg->arg.ccs_line[1].current; 
@@ -372,6 +378,8 @@ ucg_int_t ucg_dev_ic_st7735_18(ucg_t *ucg, ucg_int_t msg, void *data)
 
 ucg_int_t ucg_ext_st7735_18(ucg_t *ucg, ucg_int_t msg, void *data)
 {
+  (void)data;
+
   switch(msg)
   {
     case UCG_MSG_DRAW_L90SE:

--- a/csrc/ucg_font.c
+++ b/csrc/ucg_font.c
@@ -37,6 +37,20 @@
 
 #include "ucg.h"
 
+static void ucg_read_font_info(ucg_font_info_t *font_info, const ucg_fntpgm_uint8_t *font);
+static int8_t ucg_font_GetLowerGDescent(const void *font);
+static uint8_t ucg_font_decode_get_unsigned_bits(ucg_font_decode_t *f, uint8_t cnt);
+static ucg_int_t ucg_font_calc_vref_center(ucg_t *ucg);
+static ucg_int_t ucg_font_calc_vref_top(ucg_t *ucg);
+static ucg_int_t ucg_font_calc_vref_bottom(ucg_t *ucg);
+static ucg_int_t ucg_font_calc_vref_font(ucg_t *ucg);
+static ucg_int_t ucg_font_draw_glyph(ucg_t *ucg, ucg_int_t x, ucg_int_t y, uint8_t dir, uint8_t encoding);
+static void ucg_UpdateRefHeight(ucg_t *ucg);
+static const uint8_t *ucg_font_get_glyph_data(ucg_t *ucg, uint8_t encoding);
+static int8_t ucg_font_decode_glyph(ucg_t *ucg, const uint8_t *glyph_data);
+static void ucg_font_decode_len(ucg_t *ucg, uint8_t len, uint8_t is_foreground);
+static int8_t ucg_font_decode_get_signed_bits(ucg_font_decode_t *f, uint8_t cnt);
+
 /* font api */
 
 /* pointer to the start adress of the glyph, points to progmem area on avr uC */
@@ -155,7 +169,7 @@ static uint16_t ucg_font_get_word(const ucg_fntpgm_uint8_t *font, uint8_t offset
 
 /*========================================================================*/
 /* new font format */
-void ucg_read_font_info(ucg_font_info_t *font_info, const ucg_fntpgm_uint8_t *font)
+static void ucg_read_font_info(ucg_font_info_t *font_info, const ucg_fntpgm_uint8_t *font)
 {
   /* offset 0 */
   font_info->glyph_cnt = ucg_font_get_byte(font, 0);
@@ -274,7 +288,7 @@ uint8_t ucg_font_GetFontEndEncoding(const void *font_arg)
   return encoding;  
 }
 
-int8_t ucg_font_GetLowerGDescent(const void *font)
+static __attribute__((unused)) int8_t ucg_font_GetLowerGDescent(const void *font)
 {
   return ucg_font_get_byte(font, 14);
 }
@@ -353,7 +367,7 @@ int8_t ucg_GetFontBBXOffY(ucg_t *ucg)
   return ucg->font_info.y_offset;		/* new font info structure */
 }
 
-uint8_t ucg_GetFontCapitalAHeight(ucg_t *ucg) UCG_NOINLINE; 
+uint8_t __attribute__((unused)) ucg_GetFontCapitlaAHeight(ucg_t *ucg) UCG_NOINLINE; 
 uint8_t ucg_GetFontCapitalAHeight(ucg_t *ucg)
 {
   // OBSOLETE return ucg_font_GetCapitalAHeight(ucg->font);
@@ -364,7 +378,7 @@ uint8_t ucg_GetFontCapitalAHeight(ucg_t *ucg)
 /* glyph handling */
 
 /* optimized */
-uint8_t ucg_font_decode_get_unsigned_bits(ucg_font_decode_t *f, uint8_t cnt) 
+static uint8_t ucg_font_decode_get_unsigned_bits(ucg_font_decode_t *f, uint8_t cnt) 
 {
   uint8_t val;
   uint8_t bit_pos = f->decode_bit_pos;
@@ -409,7 +423,7 @@ uint8_t ucg_font_decode_get_unsigned_bits(ucg_font_decode_t *f, uint8_t cnt)
 
 */
 /* optimized */
-int8_t ucg_font_decode_get_signed_bits(ucg_font_decode_t *f, uint8_t cnt)
+static int8_t ucg_font_decode_get_signed_bits(ucg_font_decode_t *f, uint8_t cnt)
 {
   int8_t v, d;
   v = (int8_t)ucg_font_decode_get_unsigned_bits(f, cnt);
@@ -559,7 +573,7 @@ void ucg_font_decode_draw_pixel(ucg_t *ucg, uint8_t lx, uint8_t ly, uint8_t cnt,
     ucg_font_decode_glyph()
 */
 /* optimized */
-void ucg_font_decode_len(ucg_t *ucg, uint8_t len, uint8_t is_foreground)
+static void ucg_font_decode_len(ucg_t *ucg, uint8_t len, uint8_t is_foreground)
 {
   uint8_t cnt;	/* total number of remaining pixels, which have to be drawn */
   uint8_t rem; 	/* remaining pixel to the right edge of the glyph */
@@ -667,7 +681,7 @@ static void ucg_font_setup_decode(ucg_t *ucg, const uint8_t *glyph_data)
     ucg_font_decode_len()
 */
 /* optimized */
-int8_t ucg_font_decode_glyph(ucg_t *ucg, const uint8_t *glyph_data)
+static int8_t ucg_font_decode_glyph(ucg_t *ucg, const uint8_t *glyph_data)
 {
   uint8_t a, b;
   int8_t x, y;
@@ -718,7 +732,7 @@ int8_t ucg_font_decode_glyph(ucg_t *ucg, const uint8_t *glyph_data)
   Return:
     Address of the glyph data or NULL, if the encoding is not avialable in the font.
 */
-const uint8_t *ucg_font_get_glyph_data(ucg_t *ucg, uint8_t encoding)
+static const uint8_t *ucg_font_get_glyph_data(ucg_t *ucg, uint8_t encoding)
 {
   const uint8_t *font = ucg->font;
   font += UCG_FONT_DATA_STRUCT_SIZE;
@@ -745,7 +759,7 @@ const uint8_t *ucg_font_get_glyph_data(ucg_t *ucg, uint8_t encoding)
   return NULL;
 }
 
-ucg_int_t ucg_font_draw_glyph(ucg_t *ucg, ucg_int_t x, ucg_int_t y, uint8_t dir, uint8_t encoding)
+static ucg_int_t ucg_font_draw_glyph(ucg_t *ucg, ucg_int_t x, ucg_int_t y, uint8_t dir, uint8_t encoding)
 {
   ucg_int_t dx = 0;
   ucg->font_decode.target_x = x;
@@ -861,7 +875,7 @@ ucg_int_t ucg_DrawString(ucg_t *ucg, ucg_int_t x, ucg_int_t y, uint8_t dir, cons
 
 /* set ascent/descent for reference point calculation */
 
-void ucg_UpdateRefHeight(ucg_t *ucg)
+static void ucg_UpdateRefHeight(ucg_t *ucg)
 {
   if ( ucg->font == NULL )
     return;
@@ -907,8 +921,10 @@ void ucg_SetFontRefHeightAll(ucg_t *ucg)
 /*===============================================*/
 /* callback procedures to correct the y position */
 
-ucg_int_t ucg_font_calc_vref_font(ucg_t *ucg)
+static ucg_int_t ucg_font_calc_vref_font(ucg_t *ucg)
 {
+  (void)ucg;
+
   return 0;
 }
 
@@ -918,7 +934,7 @@ void ucg_SetFontPosBaseline(ucg_t *ucg)
 }
 
 
-ucg_int_t ucg_font_calc_vref_bottom(ucg_t *ucg)
+static ucg_int_t ucg_font_calc_vref_bottom(ucg_t *ucg)
 {
   return (ucg_int_t)(ucg->font_ref_descent);
 }
@@ -928,7 +944,7 @@ void ucg_SetFontPosBottom(ucg_t *ucg)
   ucg->font_calc_vref = ucg_font_calc_vref_bottom;
 }
 
-ucg_int_t ucg_font_calc_vref_top(ucg_t *ucg)
+static ucg_int_t ucg_font_calc_vref_top(ucg_t *ucg)
 {
   ucg_int_t tmp;
   /* reference pos is one pixel above the upper edge of the reference glyph */
@@ -942,7 +958,7 @@ void ucg_SetFontPosTop(ucg_t *ucg)
   ucg->font_calc_vref = ucg_font_calc_vref_top;
 }
 
-ucg_int_t ucg_font_calc_vref_center(ucg_t *ucg)
+static ucg_int_t ucg_font_calc_vref_center(ucg_t *ucg)
 {
   int8_t tmp;
   tmp = ucg->font_ref_ascent;

--- a/csrc/ucg_init.c
+++ b/csrc/ucg_init.c
@@ -41,8 +41,9 @@
 uint8_t global_SREG_backup;		// used by the atomic macros
 #endif
 
+static void ucg_init_struct(ucg_t *ucg);
 
-void ucg_init_struct(ucg_t *ucg)
+static void ucg_init_struct(ucg_t *ucg)
 {
   //memset(ucg, 0, sizeof(ucg_t));
   ucg->is_power_up = 0;

--- a/csrc/ucg_polygon.c
+++ b/csrc/ucg_polygon.c
@@ -276,6 +276,8 @@ void pg_ClearPolygonXY(pg_struct *pg)
 
 void pg_AddPolygonXY(pg_struct *pg, ucg_t *ucg, int16_t x, int16_t y)
 {
+  (void)ucg;
+
   if ( pg->cnt < PG_MAX_POINTS )
   {
     pg->list[pg->cnt].x = x;

--- a/csrc/ucg_rotate.c
+++ b/csrc/ucg_rotate.c
@@ -38,6 +38,10 @@
 #include "ucg.h"
 #include <assert.h>
 
+static ucg_int_t ucg_dev_rotate90(ucg_t *ucg, ucg_int_t msg, void *data);
+static ucg_int_t ucg_dev_rotate180(ucg_t *ucg, ucg_int_t msg, void *data);
+static ucg_int_t ucg_dev_rotate270(ucg_t *ucg, ucg_int_t msg, void *data);
+
 /* Side-Effects: Update dimension and reset clip range to max */
 void ucg_UndoRotate(ucg_t *ucg)
 {
@@ -65,7 +69,7 @@ static void ucg_rotate_90_xy(ucg_xy_t *xy, ucg_int_t display_width)
   
 }
 
-ucg_int_t ucg_dev_rotate90(ucg_t *ucg, ucg_int_t msg, void *data)
+static ucg_int_t ucg_dev_rotate90(ucg_t *ucg, ucg_int_t msg, void *data)
 {
   switch(msg)
   {
@@ -146,7 +150,7 @@ static void ucg_rotate_180_xy(ucg_t *ucg, ucg_xy_t *xy)
   
 }
 
-ucg_int_t ucg_dev_rotate180(ucg_t *ucg, ucg_int_t msg, void *data)
+static ucg_int_t ucg_dev_rotate180(ucg_t *ucg, ucg_int_t msg, void *data)
 {
   switch(msg)
   {
@@ -205,7 +209,7 @@ static void ucg_rotate_270_xy(ucg_t *ucg, ucg_xy_t *xy)
     xy->x = x;  
 }
 
-ucg_int_t ucg_dev_rotate270(ucg_t *ucg, ucg_int_t msg, void *data)
+static ucg_int_t ucg_dev_rotate270(ucg_t *ucg, ucg_int_t msg, void *data)
 {
   switch(msg)
   {

--- a/csrc/ucg_scale.c
+++ b/csrc/ucg_scale.c
@@ -35,6 +35,8 @@
 
 #include "ucg.h"
 
+static ucg_int_t ucg_dev_scale2x2(ucg_t *ucg, ucg_int_t msg, void *data);
+
 void ucg_UndoScale(ucg_t *ucg)
 {
   if ( ucg->scale_chain_device_cb != NULL )
@@ -49,7 +51,7 @@ void ucg_UndoScale(ucg_t *ucg)
 const ucg_fntpgm_uint8_t ucg_scale_2x2[16] UCG_FONT_SECTION("ucg_scale_2x2") =  
 { 0x00, 0x03, 0x0c, 0x0f, 0x30, 0x33, 0x3c, 0x3f, 0xc0, 0xc3, 0xcc, 0xcf, 0xf0, 0xf3, 0xfc, 0xff };
 
- static void ucg_scale_2x2_send_next_half_byte(ucg_t *ucg, ucg_xy_t *xy, ucg_int_t msg, ucg_int_t len, ucg_int_t dir, uint8_t b)
+static __attribute__((unused)) void ucg_scale_2x2_send_next_half_byte(ucg_t *ucg, ucg_xy_t *xy, ucg_int_t msg, ucg_int_t len, ucg_int_t dir, uint8_t b)
 {
   b &= 15;
   len *=2;
@@ -93,7 +95,7 @@ const ucg_fntpgm_uint8_t ucg_scale_2x2[16] UCG_FONT_SECTION("ucg_scale_2x2") =
   
 }
 
-ucg_int_t ucg_dev_scale2x2(ucg_t *ucg, ucg_int_t msg, void *data)
+static ucg_int_t ucg_dev_scale2x2(ucg_t *ucg, ucg_int_t msg, void *data)
 {
   ucg_xy_t xy;
   ucg_int_t len;

